### PR TITLE
[rtaudio] update to 6.0.1

### LINKED ISSUE #35504: [rtaudio] update to 6.0.1
### Library name

rtaudio

### New version number

6.0.1

### Other information that may be useful (release notes, etc...)

https://github.com/thestk/rtaudio/releases/tag/6.0.1

### PR DESCRIPTION
**Pull request by Cheney-W**
Fixes https://github.com/microsoft/vcpkg/issues/35504

```
Features passed with following triplets:
x86-windows 
x64-windows 
x64-windows-static 
```

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/rtaudio/fix-alsa.patch [hunk 1]
```diff
@@ -12,8 +12,8 @@ index a59e8bb..1334f5a 100644
      message(FATAL_ERROR "ALSA API requested but no ALSA dev libraries found")
    endif()
    list(APPEND INCDIRS ${ALSA_INCLUDE_DIR})
--  list(APPEND LINKLIBS ${ALSA_LIBRARY})
-+  list(APPEND LINKLIBS ${ALSA_LIBRARY} dl)
+-  list(APPEND LINKLIBS ${ALSA_LIBRARIES})
++  list(APPEND LINKLIBS ${ALSA_LIBRARIES} dl)
```
[the remark this answers is not in the document]
Comment (Cheney-W) [reply ports/rtaudio/fix-alsa.patch:16]: Thanks for point it out.

I tested in linux without the patch and the result of `ALSA_LIBRARIES` as below:
optimized;/home/vxincwa/vcpkg/installed/x64-linux/lib/libasound.a;debug;/home/vxincwa/vcpkg/installed/x64-linux/debug/lib/libasound.a;m;**dl**;Threads::Threads;rt

So I think it's not necessary anymore.

[the remark this answers is not in the document]
Comment (Cheney-W) [reply ports/rtaudio/fix-alsa.patch]: No, I don't think so. 
The original purpose of this patch was twofold: first, to add `REQUIRED` for `find_package(ALSA)`, and second, to add `dl` to `LINKLIBS`. 
However, based on the content of the [CMakeLists.txt ](https://github.com/thestk/rtaudio/blob/b4f04903312e0e0efffbe77655172e0f060dc085/CMakeLists.txt#L123C1-L123C1)file in the new version of the source:
```
# ALSA
if (RTAUDIO_API_ALSA)
  set(NEED_PTHREAD ON)
  find_package(ALSA)
  if (NOT ALSA_FOUND)
    message(FATAL_ERROR "ALSA API requested but no ALSA dev libraries found")
  endif()
  list(APPEND INCDIRS ${ALSA_INCLUDE_DIR})
  list(APPEND LINKLIBS ${ALSA_LIBRARIES})
  list(APPEND PKGCONFIG_REQUIRES "alsa")
  list(APPEND API_DEFS "-D__LINUX_ALSA__")
  list(APPEND API_LIST "alsa")
endif()
```
both of these modifications are now unnecessary. Therefore, the patch can be removed.

[the remark this answers is not in the document]
Comment (dg0yt) [reply ports/rtaudio/fix-alsa.patch]: Indeed. Upstream did the right thing. Thanks for clarification.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**